internal/dcache/file_manager/worker.go

```diff
@@ -104,11 +105,21 @@ func (wp *workerPool) queueWork(file *DcacheFile, chunk *StagedChunk, get_chunk
 
 func (wp *workerPool) readChunk(task *task) {
 	log.Info("DistributedCache::readChunk : Reading chunk idx : %d, file: %s", task.chunk.Idx, task.file.FileMetadata.Filename)
-	var err error
-
 	// Read From the Dcache
-	//Call MvRead method for reading the chunk.
-	// err = rm.MVRead()
+	readMVReq := &rm.ReadMvRequest{
+		FileID:         task.file.FileMetadata.FileID,
+		MvName:         getMVForChunk(task.chunk, &task.file.FileMetadata.FileLayout),
+		ChunkIndex:     task.chunk.Idx,
+		OffsetInChunk:  task.chunk.Idx * task.file.FileMetadata.FileLayout.ChunkSize,
+		Length:         task.chunk.Len,
+		ChunkSizeInMiB: int64(len(task.chunk.Buf)),
+		Data:           task.chunk.Buf,
+	}
+
+	readMVresp, err := rm.ReadMV(readMVReq)
+	common.Assert(err == nil)
+	common.Assert(readMVresp.BytesRead == task.chunk.Len)
```

Comment:
move this assert inside "err == nil" block below, at L124

then move the err==nil assert above after the log at L128. That way if it fails we get the error log also, which will help debugging.